contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -114,14 +144,6 @@ contract TransparentUpgradeableProxy is ERC1967Proxy {
         return _getAdmin();
     }
 
-    /**
-     * @dev Makes sure the admin cannot access the fallback function. See {Proxy-_beforeFallback}.
-     */
-    function _beforeFallback() internal virtual override {
```

Comment:
And what do you think about keeping this? It's not necessary to remove it, and by removing it we are changing the semantics of `_fallback`.

Reply:
Then it raises the question of what we do in `_fallback`.

- currently, in `_fallback` we have an
```
if (msg.sender == _getAdmin()) {
   ...
} else {
    super._fallback(); // here we should probably revert with "TransparentUpgradeableProxy: admin cannot fallback to proxy target"
}
```

- If we put the check in `_beforeFallback` and also keep it in `_fallback`, we are wasting an sload. That is 100gas on every call that goes through.
- If we put the check in `_beforeFallback` and remove it from `_fallback`, we are safe ... until someone overrides `_beforeFallback` without calling super and removes it. 

Note that the second option is already possible today. If someone overrides `_beforeFallback` and does not call super, the function selector that are not implemented in the proxy proxy become available to the admin. Is that something we want to keep ?

Reply:
That causes an inefficiency. 

For all "transparent" calls by users, here is what happens:
- `Proxy::fallback()` calls `_fallback()`
  - `TransparentUpgradeableProxy::_fallback()` (overidden) check the caller. It is not the admin, so call `super`
  - `Proxy::_fallback()` calls `_beforeFallback()`
    - `TransparentUpgradeableProxy::_beforeFallback()` (overidden) check the caller. If it is the admin we revert
    - `Proxy::_beforeFallback()` is empty
  - `Proxy::_fallback()` does the delegation

my issue is that `msg.sender != _getAdmin()` is checked twice: once in `TransparentUpgradeableProxy::_fallback()` and once in `TransparentUpgradeableProxy::_beforeCallback()`. This extra sload cost 100gas on ALL the user calls.